src/main.rs

```diff
@@ -1,18 +1,48 @@
+use std::array::TryFromSliceError;
+use std::collections::HashMap;
+use std::convert::TryInto;
 use std::io::Write;
 use std::io::{stdin, stdout};
+use std::iter::repeat;
+use std::str::from_utf8;
+
+const PAGE_SIZE: usize = 4096;
+const EMAIL_SIZE: usize = std::mem::size_of::<[u8; 255]>();
+const USERNAME_SIZE: usize = std::mem::size_of::<[u8; 32]>();
+const ID_SIZE: usize = std::mem::size_of::<u32>();
+const ROW_SIZE: usize = ID_SIZE + USERNAME_SIZE + EMAIL_SIZE;
+const ROWS_PER_PAGE: usize = PAGE_SIZE / ROW_SIZE;
+const ID_OFFSET: usize = 0;
+const USERNAME_OFFSET: usize = ID_OFFSET + ID_SIZE;
+const EMAIL_OFFSET: usize = USERNAME_OFFSET + USERNAME_SIZE;
+// const TABLE_MAX_PAGES: usize = 100;
+// const TABLE_MAX_ROWS: usize = ROWS_PER_PAGE * TABLE_MAX_PAGES;
 
 enum MetaCommand {
     Exit,
     Unsupported,
 }
 
-enum StatementType {
-    Insert,
+enum Statement<'a> {
+    Insert { row: Row<'a> },
     Select,
 }
 
-struct Statement {
-    statement_type: StatementType,
+#[derive(Debug)]
+struct Row<'a> {
+    id: u32,
+    username: &'a [u8],
```

Comment:
I chose a slice of bytes instead of an array of chars like in the original C implementation of the db tutorial, because chars in Rust and C aren't the same. I believe sing bytes here is closer to the original implementation and wanted to keep it that way in case the original authors choices are meaningful in later portions of the book.